pdf-cdn.md

```diff
@@ -0,0 +1,213 @@
+# Proposal for PDF CDN
+
+This is a proposal for how to serve PDFs from Google's CDN. This
+proposal will cover how to arrange the PDF files in Google storage,
+the load balancer, any services and the CDN to support the goals.
+
+## Goals
+1. Favor using Google products over writing code
+1. The PDF for the current version is the hot path
+1. Avoid redundant copies of the PDFs in GS
+1. Serve current version at `https://arxiv.org/pdf/2001.00021.pdf`
+1. Serve other versions at `https://arxiv.org/pdf/2001.00021v3.pdf`
+1. Maintain long standing URL patterns
+1. Not excessive time to download PDFs
+
+The only goals we can relax are 1. favor google products, 3. avoid
+redundant copies. We could relax 6 maintain URL patterns but that is
+probably a policy decision. We could relax 7 but it doesn't seem too
+relevant.
+
+## Considerations
+### Current version
+There are two main kinds of URLs we need to handle,
+the current version URLs and the versioned URLs. The current version
+URLs have filenames like `1234.12345.pdf` that lack the vN version
+number. These are expected to provide the user with the most current
+version of that paper. This can return different content at different
+times due to revisions to the paper. These revisions are not more than
+once a day. The versioned have a vN like `1234.12345v3.pdf` and are
+expected to return the PDF for that version. These are expected to
+almost never change.
+
+### Traffic
+Looking at a 1h window on 2022-10-28. Successful PDF
+requests with version number: 3766 Successful PDF requests without
+version number: 42421.
+
+### Throughput
+427000 bytes/s for successful PDFs over a 1h window on 2022-10-28.
+
+### Revisions
+Over the whole corpus about 50% of papers have
+revisions. Over the month of 2022-11 about 8% have revisions.
```

Comment:
"About 8% of the articles submitted in 2022-10 had revisions during the same month."